### PR TITLE
Proper description of the homework 4 task in the README.md file

### PR DESCRIPTION
**Pull request by ellmaster**
In order to improve readability and to unify the location of task and homework there were changes made to the README.md file.

It does cover the simplified task and examples of tests that have to be passed in order to get the homework done label.

### CODE REVIEW

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 50
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4886669&bid=16187557)

File: README.md [hunk 1]
```diff
@@ -1,77 +1,92 @@
...
+* Be careful when it comes to the BMI. You have to use cm and m for the height.
+* Do not hesitate to ask mentors for advice. This homework IS puzzling.
+
+## Task
+Modify your old functions to fit the specification:
+1) Validate number input:
++ Input is not a number -> return -1.
++ Input is a number but 0 -> return 0.
++ Input is valid -> return input.
+2) Validate name input:
```
Comment (Almantask) [inline README.md]: Codacy Issue found: [[list-item-spacing] Missing new line after list item](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4886669)

File: README.md [hunk 2]
```diff
@@ -1,77 +1,92 @@
...
+* Do use [CultureInfo.InvariantCulture](https://docs.microsoft.com/de-de/dotnet/api/system.globalization.cultureinfo.invariantculture?view=netcore-3.1) when reading in float numbers. (Due to differences in parsing numbers with '.' and ',' character.
+* The better your design, the easier it will be to find and fix errors!
+* Be careful when it comes to the BMI. You have to use cm and m for the height.
+* Do not hesitate to ask mentors for advice. This homework IS puzzling.
+
+## Task
+Modify your old functions to fit the specification:
+1) Validate number input:
++ Input is not a number -> return -1.
++ Input is a number but 0 -> return 0.
```
Comment (Almantask) [inline README.md]: Codacy Issue found: [[unordered-list-marker-style] Marker style should be `*`](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4886669)

File: README.md [hunk 3]
```diff
@@ -1,77 +1,92 @@
...
++ Input is valid -> return input.
```
Comment (Almantask) [inline README.md]: Codacy Issue found: [[unordered-list-marker-style] Marker style should be `*`](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4886669)

File: README.md [hunk 4]
```diff
@@ -1,77 +1,92 @@
...
+2) Validate name input:
++ Empty string for name -> return "-".
++ Print error message "Name cannot be empty." in a new line.
+3) Validate BMI input:
++ Height and/or weight being less or equal 0 -> return -1.
```
Comment (Almantask) [inline README.md]: Codacy Issue found: [[list-item-spacing] Missing new line after list item](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4886669)

File: README.md [hunk 5]
```diff
@@ -1,77 +1,92 @@
...
++ Print error messages:
+  - "Failed calculating BMI. Reason:"
+  - "Height cannot be equal or less than zero, but was X.", where X is the console input (X <= 0)
+  - "Weight cannot be equal or less than zero, but was X.", where X is the console input (X <= 0)
++ If height and weight are invalid print both messages, one in a new line each.
```
Comment (Almantask) [inline README.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4886669)

File: README.md [hunk 6]
```diff
@@ -1,77 +1,92 @@
...
+* Be very careful where to place your NewLine's and where not!
+* Do use [CultureInfo.InvariantCulture](https://docs.microsoft.com/de-de/dotnet/api/system.globalization.cultureinfo.invariantculture?view=netcore-3.1) when reading in float numbers. (Due to differences in parsing numbers with '.' and ',' character.
+* The better your design, the easier it will be to find and fix errors!
+* Be careful when it comes to the BMI. You have to use cm and m for the height.
+* Do not hesitate to ask mentors for advice. This homework IS puzzling.
+
+## Task
+Modify your old functions to fit the specification:
+1) Validate number input:
++ Input is not a number -> return -1.
```
Comment (Almantask) [inline README.md]: Codacy Issue found: [[unordered-list-marker-style] Marker style should be `*`](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4886669)

File: README.md (hunk 5 above)
Comment (Almantask) [inline README.md]: Codacy Issue found: [[unordered-list-marker-style] Marker style should be `*`](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4886669)

File: README.md [hunk 7]
```diff
@@ -1,77 +1,92 @@
...
++ Input is a number but 0 -> return 0.
++ Input is valid -> return input.
+2) Validate name input:
++ Empty string for name -> return "-".
++ Print error message "Name cannot be empty." in a new line.
+3) Validate BMI input:
++ Height and/or weight being less or equal 0 -> return -1.
++ Print error messages:
+  - "Failed calculating BMI. Reason:"
+  - "Height cannot be equal or less than zero, but was X.", where X is the console input (X <= 0)
```
Comment (Almantask) [inline README.md]: Codacy Issue found: [[unordered-list-marker-style] Marker style should be `*`](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4886669)

File: README.md [hunk 8]
```diff
@@ -1,77 +1,92 @@
...
+The test results should be as pointed out below.
+
+A stands for a valid value.
+X, Y stand for invalid values.
+R stands for valid value but not in the wanted range.
+E stand for empty input string. (Just pressing enter when facing prompt.)
+M stands for string message.
+
+Format:
+* Input -> return value
```
Comment (Almantask) [inline README.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4886669)

File: README.md (hunk 6 above)
Comment (Almantask) [inline README.md]: Codacy Issue found: [[list-item-spacing] Missing new line after list item](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4886669)

File: README.md [hunk 9]
```diff
@@ -1,77 +1,92 @@
...
+  * Output on same line as input.
+* ~> User/Test input being A if valid, X if invalid.
```
Comment (Almantask) [inline README.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4886669)

File: README.md [hunk 10]
```diff
@@ -1,77 +1,92 @@
...
++ Input is valid -> return input.
+2) Validate name input:
++ Empty string for name -> return "-".
++ Print error message "Name cannot be empty." in a new line.
+3) Validate BMI input:
++ Height and/or weight being less or equal 0 -> return -1.
++ Print error messages:
+  - "Failed calculating BMI. Reason:"
+  - "Height cannot be equal or less than zero, but was X.", where X is the console input (X <= 0)
+  - "Weight cannot be equal or less than zero, but was X.", where X is the console input (X <= 0)
```
Comment (Almantask) [inline README.md]: Codacy Issue found: [[unordered-list-marker-style] Marker style should be `*`](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4886669)

File: README.md [hunk 11]
```diff
@@ -1,77 +1,92 @@
...
+1) Validate number input:
++ Input is not a number -> return -1.
++ Input is a number but 0 -> return 0.
++ Input is valid -> return input.
+2) Validate name input:
++ Empty string for name -> return "-".
++ Print error message "Name cannot be empty." in a new line.
+3) Validate BMI input:
++ Height and/or weight being less or equal 0 -> return -1.
++ Print error messages:
```
Comment (Almantask) [inline README.md]: Codacy Issue found: [[unordered-list-marker-style] Marker style should be `*`](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4886669)

File: README.md [hunk 12]
```diff
@@ -1,77 +1,92 @@
...
- 
-
- 
 
+# Homework 4
+
+## Hints
+* Do use Environment.NewLine if you want to concanate a string with a newline command (or use WriteLine).
+* Be very careful where to place your NewLine's and where not!
+* Do use [CultureInfo.InvariantCulture](https://docs.microsoft.com/de-de/dotnet/api/system.globalization.cultureinfo.invariantculture?view=netcore-3.1) when reading in float numbers. (Due to differences in parsing numbers with '.' and ',' character.
```
Comment (Almantask) [inline README.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4886669)

File: README.md (hunk 11 above)
Comment (Almantask) [inline README.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4886669)

File: README.md (hunk 7 above)
Comment (Almantask) [inline README.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4886669)

File: README.md [hunk 13]
```diff
@@ -1,77 +1,92 @@
...
+* "Height cannot be equal or less than zero, but was X."
+* ""
+
+Checks.CalculateBmi(X, A) -> -1
+* "Failed calculating BMI. Reason:"
+* "Weight cannot be equal or less than zero, but was X."
+* ""
+
+Checks.CalculateBmi(X, Y) -> -1
+* "Failed calculating BMI. Reason:"
```
Comment (Almantask) [inline README.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4886669)

File: README.md [hunk 14]
```diff
@@ -1,77 +1,92 @@
...
+* Do not hesitate to ask mentors for advice. This homework IS puzzling.
+
+## Task
+Modify your old functions to fit the specification:
+1) Validate number input:
++ Input is not a number -> return -1.
++ Input is a number but 0 -> return 0.
++ Input is valid -> return input.
+2) Validate name input:
++ Empty string for name -> return "-".
```
Comment (Almantask) [inline README.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4886669)

File: README.md [hunk 15]
```diff
@@ -1,77 +1,92 @@
...
+
+A stands for a valid value.
+X, Y stand for invalid values.
+R stands for valid value but not in the wanted range.
+E stand for empty input string. (Just pressing enter when facing prompt.)
+M stands for string message.
+
+Format:
+* Input -> return value
+  * Output on same line as input.
```
Comment (Almantask) [inline README.md]: Codacy Issue found: [[list-item-spacing] Missing new line after list item](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4886669)

File: README.md (hunk 4 above)
Comment (Almantask) [inline README.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4886669)

File: README.md [hunk 16]
```diff
@@ -1,77 +1,92 @@
...
+* ~> User/Test input being A if valid, X if invalid.
+* Output2 on new line.
+  * Output3 on same line as Output2.
+
+Checks.CalculateBmi(A, X) -> -1
+* "Failed calculating BMI. Reason:"
+* "Height cannot be equal or less than zero, but was X."
```
Comment (Almantask) [inline README.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4886669)

File: README.md [hunk 17]
```diff
@@ -1,77 +1,92 @@
...
+
+## Task
+Modify your old functions to fit the specification:
+1) Validate number input:
++ Input is not a number -> return -1.
++ Input is a number but 0 -> return 0.
++ Input is valid -> return input.
+2) Validate name input:
++ Empty string for name -> return "-".
++ Print error message "Name cannot be empty." in a new line.
```
Comment (Almantask) [inline README.md]: Codacy Issue found: [[unordered-list-marker-style] Marker style should be `*`](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4886669)

File: README.md (hunk 9 above)
Comment (Almantask) [inline README.md]: Codacy Issue found: [[list-item-spacing] Missing new line after list item](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4886669)

File: README.md (hunk 2 above)
Comment (Almantask) [inline README.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4886669)

File: README.md (hunk 17 above)
Comment (Almantask) [inline README.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4886669)

**Comment by ellmaster** [comment]
> I am a bit overwhelmed with all the details, maybe too much about tests?

I had that thought, too. On the other hand I wanted this to be like a reference where you can come back to and look at the specifications of what is tested. 

Maybe we can invite some students to look at it and request changes because it is primarily for them?

**Pull request merged**